src/Store/StoreThread.re

```diff
@@ -80,17 +82,15 @@ let start =
       ~setTitle,
       ~setVsync,
       ~window: option(Revery.Window.t),
-      ~cliOptions: option(Oni_Core.Cli.t),
+      ~filesToOpen=[],
+      ~overriddenExtensionsDir=?,
+      ~shouldLoadExtensions=true,
+      ~shouldSyntaxHighlight=true,
+      ~shouldLoadConfiguration=true,
```

Comment:
Switched form passing around the whole `cliOptions` record in favor of explicit arguments. This gives more control over the options, a better udnerstanding of the dependencies, and allowed the `Cli` module to be moved to `bin_editor` where CLI parsing and such belongs.